ilastik/shell/gui/ilastikShell.py

```diff
@@ -396,6 +396,51 @@ def handleAppletProgressImpl(self, index: int, percentage: int, cancelled: bool)
 """
 
 
+class StartupContainer(QWidget):
+    """Container widget for startup page that allows drag-n-dropping project files.
+
+    included via `ilastik/shell/gui/ui/ilastikShell.ui`
+    """
+
+    ilpDropped = pyqtSignal(str)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.setAcceptDrops(True)
+
+    def dropEvent(self, a0):
+        if a0 is None:
+            return
+        mimedata = a0.mimeData()
+        if mimedata is None:
+            return
+        urls = mimedata.urls()
+        assert len(urls) == 1, f"Drag and drop should only ever get a single path, got {len(urls)}"
+        url = urls[0].toLocalFile()
+        assert url.endswith(
+            ".ilp"
+        ), f"Drag and drop should should only ever receive files that end in '.ilp', got {url}"
+        self.ilpDropped.emit(url)
+
+    def dragEnterEvent(self, a0):
+        if a0 is None:
+            return
+        mimedata = a0.mimeData()
+        if mimedata is None:
+            return
+        # Only accept drag-and-drop events that consist of urls to local .ilp files.
+        if not mimedata.hasUrls():
+            return
+        urls = mimedata.urls()
+
+        if len(urls) != 1:
```

Comment:
Wouldn't it be more convenient/forgiving to allow dropping multiple items, and just pick the first `.ilp` from the list? Maybe even allow to drop a folder, and search for a project file inside.

I could see this happening if e.g. someone likes to drag-select things in a file manager instead of just clicking.

Obviously, the tradeoff here is a potential confusion "What project did I open?.."

Reply:
Hello @emilmelnikov, first of all thank you for looking into this!!

After reading your comment I thought about it and I would prefer this to be very predictable. If you drop a folder with multiple project files what would you expect? Would you follow the full tree?

The question is whether something more helpful should happen than what is implemented here (nothing, will not accept unsuitable drops). One could still take the drop and inform the user that only a single `.ilp` file can be dropped?

Reply:
Yes, this is probably the best solution, since ilastik is used to analyze "serious" data. What if someone has two similar projects nearby and opens the wrong one?..

I've suggested this because I've seen some desktop apps allow you to DnD any files into a resource manager window, and adds only those that it recognizes.

Reply:
Yes, I thought about different scenarios, too, the one you mention included. But this I expect only from applications that somehow have the concept of many files. For the softwares that have the concept of "projects" I wouldn't expect them to handle them. But yes, I'll warn if there were more than one project file :)